utils/firestore.js

```diff
@@ -1637,6 +1628,50 @@ const storeNotifications = async payload => {
     }
 }
 
+/**
+ * Save notification records to `notifications` collection.
+ * @param {object[]} notificationRecordArray Array of notification records to be saved
+ */
+const saveNotificationBatch = async (notificationRecordArray) => {
+  const chunkSize = 500; // batched write has a doc limit of 500
+  const chunkArray = createChunkArray(notificationRecordArray, chunkSize);
+  for (const recordChunk of chunkArray) {
+    const batch = db.batch();
+    try {
+        for (const record of recordChunk) {
+        const docId = record.uid + record.notificationSpecificationsID.slice(0, 6);
```

Comment:
is it necessary to create our own doc id?

Reply:
When using a random ID, each document has a unique ID and is treated as unique, even some documents are identical in contents (eg duplicate emails saved). With a ID generated from uid and spec ID, each notification (to a person) will have only one copy. When a message is saved multiple times, later saves cover earlier saves.